app.js

```diff
@@ -0,0 +1,84 @@
+var path = require('path');
+var fs = require('fs');
+var express = require('express');
+var Router = require('react-router');
+
+var indexStaticWatcher = require('./lib/index-static-watcher').create();
+var PORT = process.env.PORT || 8008;
+var PRODUCTION = (process.env.NODE_ENV === 'production');
+var DIST_DIR = path.join(__dirname, 'dist');
+
+var indexStatic;
+var router;
+var app = express();
+
+var startProdApp = function() {
+  console.log([
+    'Production mode enabled. Note that "npm install" is assumed to',
+    'have recently been run with NODE_ENV="production". If this is not',
+    'the case, some or all static assets may be out of date.'
+  ].join('\n'));
+  indexStaticWatcher.build(function(err, newIndexStatic) {
+    if (err) {
+      throw err;
+    }
+
+    console.log('Built server-side bundle.');
```

Comment:
> 1. `app.bundle.js` (what's this? I can't find out how this is generated from)

This is basically the main entry point for the website. It's generated from this part of our `webpack.config.js`:

```javascript
module.exports = {
  entry: {
    app: './lib/main.jsx',
    manualTests: './test/browser/manual-main.jsx',
    tests: './test/browser/main.js'
  },
```

Basically webpack is generating three different entry points for our code, to be used by three different kinds of pages: the "app", which is loaded by any public page on our site, the automated test suite, and the manual test suite. In the future, we might add more.

(Note also that this file isn't new to this PR, it's been around since the early days of this project.)

> 2. `commons.bundle.js` (I guess this is from [here](https://github.com/mozilla/teach.webmaker.org/blob/develop/webpack.config.js#L53))?

Yup, it's our [commons chunk](http://webpack.github.io/docs/list-of-plugins.html#commonschunkplugin), which contains common modules shared between all our entry points.

(Note also that this file isn't new to this PR, it's been around since the early days of this project.)

> 3. `index-static.bundle.js`
>
> Are the first 2 for browser and `index-static.bundle.js` for server to watch for code changes & generate `/dist`?

Yup exactly! Well, mostly exactly... `index-static.bundle.js` was introduced in #839 and is used to generate the purely static site, but it's *also* being leveraged by the lightweight dynamic server in this PR, to dynamically generate the HTML pages.

Another way to say it is that the website's HTML pages can *either* be statically generated in `dist` *or* dynamically served, and `index-static.bundle.js` is required to do both.